godtossing/Pos/Ssc/GodTossing/Seed.hs

```diff
@@ -6,38 +8,40 @@ module Pos.Ssc.GodTossing.Seed
 
 import           Universum
 
-import           Control.Lens                 (_Left)
-import qualified Data.HashMap.Strict          as HM
-import qualified Data.HashSet                 as HS
-
-import           Pos.Binary.Class             (fromBinary, fromBinaryM)
-import           Pos.Core                     (SharedSeed, StakeholderId, addressHash,
-                                               mkCoin, sumCoins, unsafeIntegerToCoin)
-import           Pos.Crypto                   (Secret, Share, unsafeRecoverSecret,
-                                               verifySecretProof)
-import           Pos.Lrc.Types                (RichmenStakes)
-import           Pos.Ssc.GodTossing.Core      (Commitment (..),
-                                               CommitmentsMap (getCommitmentsMap),
-                                               OpeningsMap, SharesMap, getOpening,
-                                               secretToSharedSeed, verifyOpening)
-import           Pos.Ssc.GodTossing.Error     (SeedError (..))
-import           Pos.Ssc.GodTossing.Functions (vssThreshold)
-import           Pos.Util.Util                (getKeys)
-
+import           Control.Lens             (_Left)
+import qualified Data.HashMap.Strict      as HM
+import qualified Data.HashSet             as HS
+
+import           Pos.Binary.Class         (AsBinary, fromBinaryM)
+import           Pos.Core                 (SharedSeed, StakeholderId, addressHash, mkCoin,
+                                           sumCoins, unsafeIntegerToCoin)
+import           Pos.Crypto               (DecShare, Secret, VssPublicKey, recoverSecret,
+                                           verifySecret)
+import           Pos.Lrc.Types            (RichmenStakes)
+import           Pos.Ssc.GodTossing.Core  (Commitment (..), CommitmentsMap (..),
+                                           Opening (..), OpeningsMap, SharesMap,
+                                           SignedCommitment, getCommShares,
+                                           secretToSharedSeed, verifyOpening,
+                                           vssThreshold)
+import           Pos.Ssc.GodTossing.Error (SeedError (..))
+import           Pos.Util.Util            (getKeys)
 
 -- | Calculate SharedSeed. SharedSeed is a random bytestring that all
 -- nodes generate together and agree on.
---
--- TODO: do we need to check secrets' lengths? Probably not.
```

Comment:
@neongreen 
> Where is this checked now?

If it's not checked, are you sure we don't need to check it?

Reply:
We don't need to check it because in SCRAPE we have

```haskell
-- | Secret
newtype Secret = Secret Point
    deriving (Show,Eq,NFData,Binary)

pointToDhSecret :: Point -> DhSecret
pointToDhSecret (Point p) =
    let (x, _) = SSL.ecPointToAffineGFp p256 p
     in DhSecret $ B.convert $ hashSHA256 $ i2ospOf_ 32 x
```

i.e. it will always have the same length and the attacker can't influence it. A secret is a SHA256 hash of math. MATH